### PR TITLE
🙅 Gracefully reject proposals and requests

### LINKED ISSUE #20288: Rejecting Wallet Connect requests and sessions
# Feature Issue

[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a user, I would like to be able to reject a session/request, so that the dapp is aware of the interaction state and the user can continue using it.

### Acceptance Criteria

- [ ] the session/request should be rejected when the user presses the `x` button in the navigation header
- [ ] the session/request should be ignored when the user dismisses the sheet by dragging it down
- [ ] the dapp should show the rejected state when the user rejects the session

### Notes

[comment]: # (For the developer who will implement. Include planning poker estimates, guidance, etc.)

### PR DESCRIPTION
**Pull request by shivekkhurana**
Fixes #20288 
Superceeds #20528

### CODE REVIEW

File: src/react_native/wallet_connect.cljs [hunk 1]
```diff
@@ -23,6 +23,8 @@
              :supportedNamespaces supported-namespaces})))
 
 (defn get-sdk-error
+  "Get an error from this list
+  https://github.com/WalletConnect/walletconnect-monorepo/blob/c6e9529418a0c81d4efcc6ac4e61f242a50b56c5/packages/utils/src/errors.ts"
```
Comment (clauxx) [inline src/react_native/wallet_connect.cljs]: Maybe we could move this to a comment. Seems weird to have this url as a docstring

**Review by clauxx — APPROVED** [review]
Looks good!

**Comment by clauxx** [comment]
@pavloburykh are the failed tests related or is it ok to merge?

**Comment by pavloburykh** [comment]
> @pavloburykh are the failed tests related or is it ok to merge?

@clauxx thanks for the PR. e2e failures are not PR related. Ready for merge.

File: src/status_im/contexts/wallet/wallet_connect/effects.cljs [hunk 2]
```diff
@@ -118,12 +118,27 @@
...
                              (clj->js {:topic    topic
-                                       :response {:id      id
-                                                  :jsonrpc "2.0"
-                                                  :result  result}}))
+                                       :response (merge {:id      id
+                                                         :jsonrpc "2.0"}
+                                                        (when result
+                                                          {:result result})
+                                                        (when error
+                                                          {:error error}))}))
```
Comment (alwx) [inline src/status_im/contexts/wallet/wallet_connect/effects.cljs:130]: why do you need `when` here?

**Review by alwx — APPROVED** [review]

**Pull request merged**